packages/firestore/src/local/memory_query_cache.ts

```diff
@@ -81,7 +92,9 @@ export class MemoryQueryCache implements QueryCache {
     if (targetId > this.highestTargetId) {
       this.highestTargetId = targetId;
     }
-    // TODO(GC): track sequence number
+    if (queryData.sequenceNumber > this.highestSequenceNumber) {
```

Comment:
You don't save the sequence number in the Android port here or in the IndexedDb version. I can't convince myself which one is correct, so I'll let you do that for me.